Composer/packages/client/src/pages/botProject/adapters/ExternalAdapterModal.tsx

```diff
@@ -35,18 +35,24 @@ type Props = {
   projectId: string;
   schema: JSONSchema7;
   uiSchema: JSONSchema7 & { helpLink?: string };
-  value?: { [key: string]: ConfigValue };
+  value?: { [key: string]: JSONSchema4Type };
 };
 
-export function hasRequired(testObject: { [key: string]: ConfigValue }, fields?: string[]) {
+export function hasRequired(testObject: { [key: string]: JSONSchema4Type | undefined }, fields?: string[]) {
   if (fields == null || fields.length === 0) return true;
   return fields.every((field: string) => field in testObject);
 }
 
+function makeDefault(schema: JSONSchema7) {
+  const { properties } = schema;
+
+  return mapValues(properties, 'default');
+}
+
 const AdapterModal = (props: Props) => {
   const { isOpen, onClose, schema, uiSchema, projectId, adapterKey, packageName, isFirstTime } = props;
 
-  const [value, setValue] = useState(props.value);
+  const [value, setValue] = useState(props.value == null ? makeDefault(schema) : props.value);
```

Comment:
A couple of questions:
1. What if the value is `null` and that is a valid value?
2. Depending on the answer to the above -- could we use the nullish operator instead?

Reply:
The only possible things `value` can be are `null` or an object with settings in it; if it's `null` then we're in the case where we tried to find settings in appSettings.json and they weren't there. That being said, it makes sense to use the coalesce operator here, so I'll do that.